lib/models/Player.js

```diff
@@ -85,12 +85,15 @@ function parseTrackMetadata(metadata, nextTrack) {
     const sax = flow(streamer(metadata.val));
 
     sax.on('tag:item', (item) => {
-      track.uri = item.res.$text;
-      track.duration = parseTime((item.res.$attrs || item.res).duration);
-      track.artist = item['dc:creator'];
-      track.album = item['upnp:album'];
-      track.title = item['dc:title'];
-      track.albumArtUri = item['upnp:albumarturi'];
+      if (item.res && item.res.$text) track.uri = item.res.$text;
+      if (item.res && (item.res.$attrs || item.res).duration) track.duration = parseTime((item.res.$attrs || item.res).duration);
+      if (item['dc:creator']) track.artist = item['dc:creator'];
+      if (item['upnp:album']) track.album = item['upnp:album'];
+      if (item['dc:title']) track.title = item['dc:title'];
+      if (item['upnp:albumarturi']) track.albumArtUri = item['upnp:albumarturi'];
+      if (item['r:resmd']) track.metaData = item['r:resmd'];
+      if (item['r:streamcontent']) track.streamInfo = item['r:streamcontent'];
+      if (item['r:radioshowmd']) track.radioShowMetaData = item['r:radioshowmd'];
     });
```

Comment:
```let track = nextTrack ? clone(EMPTY_STATE.nextTrack) : clone(EMPTY_STATE.currentTrack);``` so we start with a clean state. If we don't check if a setter is necessary, things like ```track.streamInfo``` or ```track.radioShowMetaData``` will be overwritten with ```undefined``` if the track is no radio. And that would make the default values defined in the ```EMPTY_STATE``` obsolete.